src/server/pkg/hashtree/db.go

```diff
@@ -1216,6 +1145,37 @@ func (mq *mergePQ) next() ([]*MergeNode, error) {
 	return ns, nil
 }
 
+func (mq *mergePQ) peek() ([]*MergeNode, error) {
```

Comment:
So, I don't think this is actually going to work when we have multiple of the same paths in the queue (which does not happen now because this operates on merged tree shards). During the traversal, we are going to break at the compare(i, next) when the next entry has the same path as well as when the next path is greater. I also think that this would not handle the case when both the left and right entry have the same path as the root. I think the "fill" operation is fundamentally different from what you want here which is essentially a level order traversal.

Reply:
Fixed.